### ISSUE TITLE
Jenkins LTS will drop support for Java 11 in October

### ISSUE THREAD

**Issue by richardlau** [nodejs/build#3916]
https://www.jenkins.io/blog/2024/06/11/require-java-17/
> The first LTS release to require Java 17 or newer will ship at the end of October 2024.

After https://github.com/nodejs/build/issues/3030 most of our agents are being run with Java 17.

Scanning through https://ci.nodejs.org/computer/ it looks like Java 11 is still being used on:
- CentOS 7
- Debian 10
- FreeBSD 12
- SmartOS
- IBM i

We can drop the CentOS 7 machine. 

@abmusse Is Java 17 available on IBM i?

**Comment by abmusse** [comment]
@richardlau 

Yes, Java 17 is available but it looks like its not packaged as an RPM.
I'm investigating what would be required to upgrade to Java 17.

**Comment by richardlau** [comment]
@nodejs/build-infra Do we want to keep Debian 10 around in the CI or just remove? It's [End-of-Life](https://wiki.debian.org/LTS) as of June 2024.

(If we want to keep it, there are options for Java 17 -- both Oracle and Adoptium offer Java 17 that should work on Debian 10 AFAICT.)

**Comment by sxa** [comment]
Debian 10 has the same glibc level as RHEL8 so it's technically feasible to keep that going as a functional system ....
However since it's out of support I'm not sure I'd want to keep it on the network, so perhaps we should just upgrade it to a later release.

**Comment by richardlau** [comment]
Another reason to remove Debian 10 is that npm 10.8.3 contains an update to node-gyp that now requires Python 3.8. The packaged version of Python on Debian 10 is 3.7.3. While it might be possible to install newer Python 3 on the machine by other means, that in combination with the Java 11 issue will add more maintenance overhead to keeping this End-of-Life platform active in the CI.

Removal of Debian 10 will still leave Debian 11 and 12 in the CI, so we'll still have coverage on Debian there.

**Comment by targos** [comment]
On the [release CI](https://ci-release.nodejs.org/computer/), I see only [release-osuosl-centos7-arm64-1](https://ci-release.nodejs.org/computer/release%2Dosuosl%2Dcentos7%2Darm64%2D1/). Can we remove that builder? I see nothing in its build history.

**Comment by richardlau** [comment]
> On the [release CI](https://ci-release.nodejs.org/computer/), I see only [release-osuosl-centos7-arm64-1](https://ci-release.nodejs.org/computer/release%2Dosuosl%2Dcentos7%2Darm64%2D1/). Can we remove that builder? I see nothing in its build history.

Yes, we don't use CentOS 7 anymore (from Node.js 18 onwards).

**Comment by richardlau** [comment]
Current status 

## ci

platform | notes
---|---
CentOS 7 | drop
Debian 10 | drop
FreeBSD 12 | Migrate to FreeBSD 13 https://github.com/nodejs/build/issues/3593
SmartOS | https://github.com/nodejs/build/issues/3731
IBM i | Requires update to IBM i 7.4 (@abmusse has requested)

## ci-release

platform | notes
---|---
CentOS 7 | drop

**Comment by richardlau** [comment]
IBM i update: plan is for the machines to be updated to IBM i 7.4 next week. After that we (@abmusse ?) will need to update Jenkins and Ansible.

**Comment by abmusse** [comment]
Yes Once the machines get updated to 7.4 we will need to do some config changes to pickup Java 17.

**Comment by abmusse** [comment]
We now have Java 17 installed on the IBM i CI machines 🎉

**Comment by targos** [comment]
CentOS 7 and Debian 10 machines were removed.
The remaining ones with Java 11 are all SmartOS:
- https://ci.nodejs.org/computer/test%2Dequinix%5Fmnx%2Dsmartos18%2Dx64%2D3/
- https://ci.nodejs.org/computer/test%2Dequinix%5Fmnx%2Dsmartos18%2Dx64%2D4/
- https://ci.nodejs.org/computer/test%2Dequinix%5Fmnx%2Dsmartos20%2Dx64%2D3/
- https://ci.nodejs.org/computer/test%2Dequinix%5Fmnx%2Dsmartos20%2Dx64%2D4/
